Lib/ssl.py

```diff
@@ -831,6 +833,14 @@ def context(self):
     def context(self, ctx):
         self._sslobj.context = ctx
 
+    @property
+    def outer_server_hostname(self) -> str:
+        """The server name used in the outer ClientHello."""
+        if self._sslobj:
+            return self._sslobj.get_ech_status()[2]
+        else:
+            raise ValueError("No SSL wrapper around " + str(self))
```

Comment:
```suggestion
    @property
    def outer_server_hostname(self):
        """The server name used in the outer ClientHello."""
        return self._sslobj.get_ech_status()[2]
```

We don't use type annotations in the standard library, it's left to typeshed. And let's just propagate an AttributeError as for the other properties.